### PR TITLE
feat: Add context manager for prompt template, version, and variables

### PR DESCRIPTION
**Pull request by fjcasti1**

### CODE REVIEW

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 1]
```diff
@@ -166,6 +184,61 @@ def __init__(self, tags: List[str]) -> None:
...
+class using_prompt_template_version(_UsingAttributesContextManager):
+    """
+    Context manager to add prompt template version to the current OpenTelemetry Context.
+    OpenInference instrumentations will read this Context and pass the prompt template
+    version as a span attribute, following the OpenInference semantic conventions.
+
+    Examples:
+        with using_prompt_template_version("v1.0"):
+            # Tracing within this block will include the span attribute:
+            # "llm.prompt_template.version" = "v1.0"
```
Comment (mikeldking) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py:215]: Let's combine all of these as they should probably all be set at the same time

**Review by mikeldking — CHANGES_REQUESTED** [review]

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py (hunk 1 above)
Comment (fjcasti1) [reply python/openinference-instrumentation/src/openinference/instrumentation/__init__.py]: Much better, what was I thinking?

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 2]
```diff
@@ -166,6 +184,44 @@ def __init__(self, tags: List[str]) -> None:
...
+            # "llm.prompt_template.variables" = "{\"city\": \"Johannesburg\",
+            \"date\": \"July 11\"}"
+            ...
+    """
+
+    def __init__(
+        self,
+        *,
+        template: str = "",
+        version: str = "",
```
Comment (mikeldking) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py:215]:
```suggestion
        version: Optional[str] = "",
```

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 3]
```diff
@@ -166,6 +184,44 @@ def __init__(self, tags: List[str]) -> None:
...
+            # "llm.prompt_template.version" = "v1.0"
+            # "llm.prompt_template.variables" = "{\"city\": \"Johannesburg\",
+            \"date\": \"July 11\"}"
+            ...
+    """
+
+    def __init__(
+        self,
+        *,
+        template: str = "",
```
Comment (mikeldking) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py:214]:
```suggestion
        template: Optional[str] = "",
```

**Review by mikeldking — APPROVED** [review]
Nice!

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 4]
```diff
@@ -192,51 +192,33 @@ class using_prompt_template(_UsingAttributesContextManager):
 
     Examples:
         prompt_template = "Please describe the weather forecast for {city} on {date}"
-        with using_prompt_template(prompt_template):
+        prompt_template_variables = "{"city": "Johannesburg", date:"July 11"}"
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py:196]:
```suggestion
        prompt_template_variables = {"city": "Johannesburg", date: "July 11"}
```

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 5]
```diff
@@ -166,6 +184,44 @@ def __init__(self, tags: List[str]) -> None:
         super().__init__(tags=tags)
 
 
+class using_prompt_template(_UsingAttributesContextManager):
+    """
+    Context manager to add prompt template, with its version and variables a to the
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py:189]:
```suggestion
    Context manager to add prompt template, with its version and variables to the
```

**Review by axiomofjoy — APPROVED** [review]
nice, i think you're using strings rather than dictionaries in the docstrings for metadata, but other than that, looks good. thanks for the thorough docstrings.

**Comment by fjcasti1** [comment]
> nice, i think you're using strings rather than dictionaries in the docstrings for metadata, but other than that, looks good. thanks for the thorough docstrings.

Good catch!

**Pull request merged**